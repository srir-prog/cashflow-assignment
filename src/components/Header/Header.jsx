import React from "react";
import logo from "./../../../src/centime_logo.png";
import { useTranslation } from "react-i18next";

const Header = () => {
  const [t, i18n] = useTranslation("common");
  return (
    <div className="top">
      <img
        className="header__logo"
        src={logo}
        alt={t("header.imgDescription")}
      ></img>
      <div className="add__locale">
        <button id="english" onClick={() => i18n.changeLanguage("en")}>
          English
        </button>
        <button id="german" onClick={() => i18n.changeLanguage("de")}>
          German
        </button>
      </div>
    </div>
  );
};

export default Header;
