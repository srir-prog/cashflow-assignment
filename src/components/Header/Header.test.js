import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from './Header';

let component;
beforeEach(() => {
    //Mock out "react-i18next" library.
    jest.mock("react-i18next");
    component = shallow(<Header />);
});
it('should render correctly', () => {
    //Snapshot testing.
    const tree = renderer.create(<Header />);
    expect(tree.toJSON()).toMatchSnapshot();
});
it('should render <img>', () => {
    const imageElement = component.find('img');
    expect(imageElement.length).toBe(1);
});
it('should render buttons', () => {
    const buttonElement = component.find('button');
    expect(buttonElement.length).toBe(2);
});
