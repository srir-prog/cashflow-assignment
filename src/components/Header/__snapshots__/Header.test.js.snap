// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="top"
>
  <img
    alt="header.imgDescription"
    className="header__logo"
    src="centime_logo.png"
  />
  <div
    className="add__locale"
  >
    <button
      onClick={[Function]}
    >
      English
    </button>
    <button
      onClick={[Function]}
    >
      German
    </button>
  </div>
</div>
`;
